pkg/kubectl/cmd/get.go

```diff
@@ -187,8 +187,11 @@ func RunGet(f cmdutil.Factory, out, errOut io.Writer, cmd *cobra.Command, args [
 	if err != nil {
 		return err
 	}
-	if len(options.Filenames) > 0 || argsHasNames {
-		cmd.Flag("show-all").Value.Set("true")
+	output := cmdutil.GetFlagString(cmd, "output")
+	if len(options.Filenames) > 0 || argsHasNames || output == "json" || output == "yaml" {
```

Comment:
What happens if `--output` is `go-template*` or `jsonpath*`? I believe it should apply to all too (no filtering). Also, I'd suggest to move the output check to a helper function like `OutputsRawFormat(output string) bool` or something similar.